### PR TITLE
create acrn hypervisor/kernel deb

### PR DESCRIPTION
**Pull request by zhanqigithub**
Tracked-On: #5022

Signed-off-by: zhanqi <sherry.qi.zhang@intel.com>

### CODE REVIEW

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

**Comment by wenlingz** [comment]
@dongyaozu  please help approve this PR, thanks

**Comment by wenlingz** [comment]
@dongyaozu please help approve this PR, thanks

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

File: misc/packaging/README.md [hunk 1]
```diff
@@ -0,0 +1,117 @@
+# ubuntuSoS
```
Comment (dbkinder) [inline misc/packaging/README.md:1]: For inclusion in our documentation, this needs to be written as restructureText as is all our other documentation.  Look at one of the other README.rst files in the misc folders.  The doc will also need to be added to the table of contents once it's ready.

File: misc/packaging/README.md [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+this scripts show how to build acrn-hypervisor and acrn-kernel deb package
+
+1. setup native ubuntu
```
Comment (dbkinder) [inline misc/packaging/README.md:4]: Which version?  18.04, 20.04?  Desktop? Server?

File: misc/packaging/README.md [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+
+install python3
```
Comment (dbkinder) [inline misc/packaging/README.md:6]: Is this the only missing package requirement?  It's best to give the commands to do the install, ``sudo apt install python3`` (and will that get all the expected python utils)?

File: misc/packaging/README.md [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+
+config proxy
+
+sudo su
```
Comment (dbkinder) [inline misc/packaging/README.md:10]: why is this command needed?

File: misc/packaging/README.md [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+
+
+2. config json file
```
Comment (dbkinder) [inline misc/packaging/README.md:13]: Configure which json file?  Are we adding the information below to some existing json file?

File: misc/packaging/README.md [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+
+
+{
+
+	"//":"release ubuntu as sos verion",
+
+	"install_package":"true",---------->this 1st time should set to true if you did not install acrn related compile package, if already install,set to false
+
+	"gcc_version":"7.3.0",---------->higher than this
```
Comment (dbkinder) [inline misc/packaging/README.md:22]: what does "higher than this" mean?

File: misc/packaging/README.md [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+
+	"binutils":"2.27",---------->higher than this
+
+	"//":"acrn-hypervisor config",
+
+	"build_acrn":"true",---------->if you need build acrn-hypervisor
```
Comment (dbkinder) [inline misc/packaging/README.md:28]: How do you know if this is needed?

File: misc/packaging/README.md [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+
+	"sync_acrn_code":"true",---------->if you need sync acrn-hypervisor code
```
Comment (dbkinder) [inline misc/packaging/README.md:30]: How do you know if this is needed?

File: misc/packaging/README.md [hunk 9]
```diff
@@ -0,0 +1,117 @@
...
+
+	"//":"acrn-hypervisor config",
+
+	"build_acrn":"true",---------->if you need build acrn-hypervisor
+
+	"sync_acrn_code":"true",---------->if you need sync acrn-hypervisor code 
+
+	"acrn_repo":"https://github.com/projectacrn/acrn-hypervisor.git",---------->acrn repo
+
+	"release_version":"remotes/origin/release_2.0",---------->acrn release branch
```
Comment (dbkinder) [inline misc/packaging/README.md:34]: the branch name is "release_2.0".  Why do I need to have the ``remotes/origin/`` before the release version name?

File: misc/packaging/README.md [hunk 10]
```diff
@@ -0,0 +1,117 @@
...
+
+	"acrn_deb_package":"true",---------->if you need create acrn-hypervison deb
```
Comment (dbkinder) [inline misc/packaging/README.md:36]: How do you know if this is needed, same for the next item?

File: misc/packaging/README.md [hunk 11]
```diff
@@ -0,0 +1,117 @@
...
+
+	"install_acrn_deb":"false",---------->if you need install acrn-hypervisor deb
+
+	"build_cmd":---------->acrn-hypervisor build command should include scenario and board info and release type
```
Comment (dbkinder) [inline misc/packaging/README.md:40]: should or must?

File: misc/packaging/README.md [hunk 12]
```diff
@@ -0,0 +1,117 @@
...
+
+	{
+
+		"scenario":
+
+		{
+
+			"industry":"true",
```
Comment (dbkinder) [inline misc/packaging/README.md:48]: can all these scenarios be set to true at the same time?

File: misc/packaging/README.md [hunk 13]
```diff
@@ -0,0 +1,117 @@
...
+			"logical_partition":"true"
+
+
+		},
+
+		"board":
+
+		{
+
+			"nuc7i7dnb":"true",
```
Comment (dbkinder) [inline misc/packaging/README.md:61]: can multiple boards be set to "true" at the same time?

File: misc/packaging/README.md [hunk 14]
```diff
@@ -0,0 +1,117 @@
...
+
+			"whl-ipc-i5":"true"
+
+		},
+
+		"release":"0"
```
Comment (dbkinder) [inline misc/packaging/README.md:67]: No explanation for what this setting is for?  Is this for "release" vs "debug"?

File: misc/packaging/README.md [hunk 15]
```diff
@@ -0,0 +1,117 @@
...
+
+	},
+
+	"//":"kernel config",
+
+	"build_acrn_kernel":"true",---------->if you need build acrn-kernel
```
Comment (dbkinder) [inline misc/packaging/README.md:73]: same questions here, how do you know if you need to set these to "true" ?

File: misc/packaging/README.md [hunk 16]
```diff
@@ -0,0 +1,117 @@
...
+	"sos_kernel_repo":"https://github.com/projectacrn/acrn-kernel.git",---------->acrn kernel repo
+
+
+	"acrn_kernel_deb_package":"true",---------->if you need create acrn-kernel deb
+
+	"install_acrn_kernel_deb":"false",---------->if you need install acrn-kernel deb
+
+	"//":"misc",
+
+	"auto_reboot":"false"---------->if you need ireboot
```
Comment (dbkinder) [inline misc/packaging/README.md:89]: what is "ireboot"?

File: misc/packaging/acrn-hypervisor.postinst [hunk 17]
```diff
@@ -0,0 +1,120 @@
+#!/bin/bash
+#* Copyright (c) 2020 Intel Corporation
```
Comment (dbkinder) [inline misc/packaging/acrn-hypervisor.postinst:2]: needs a license tag comment

File: misc/packaging/acrn-hypervisor.postinst [hunk 18]
```diff
@@ -0,0 +1,120 @@
...
+# postinst script for acrn-hypervisor
+set -e
+
+
+echo "please choose <scenario> ,<board> ,<disk type>"
+
+echo "Senario is ->
```
Comment (dbkinder) [inline misc/packaging/acrn-hypervisor.postinst:9]: "scenario" is misspelled multiple times in this script

File: misc/packaging/acrn-hypervisor.postinst [hunk 19]
```diff
@@ -0,0 +1,120 @@
...
+        echo "Senario is hybrid"
+        SCENARIO="hybrid"
+
+elif [[ $num -eq 3 ]]
+then
+        echo "Senario is logical_partition"
+        SCENARIO="logical_partition"
+fi
+
+echo "Board is ->
```
Comment (dbkinder) [inline misc/packaging/acrn-hypervisor.postinst:30]: These are the only two supported boards?  What about all the other supported hardware we list?

File: misc/packaging/acrn-hypervisor.postinst [hunk 20]
```diff
@@ -0,0 +1,120 @@
...
+
+echo "please choose <scenario> ,<board> ,<disk type>"
+
+echo "Senario is ->
+        1. industry
+        2. hybrid
+        3. logical_partition"
+read num
+
+if [[ $num -eq 1 ]]
```
Comment (dbkinder) [inline misc/packaging/acrn-hypervisor.postinst:15]: Needs some error checking (what if I pick 4?)

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

File: misc/packaging/README.md (hunk 1 above)
Comment (zhanqigithub) [reply misc/packaging/README.md]: change to rst

File: misc/packaging/README.md (hunk 3 above)
Comment (zhanqigithub) [reply misc/packaging/README.md]: add sudo apt install python3

File: misc/packaging/README.md (hunk 5 above)
Comment (zhanqigithub) [reply misc/packaging/README.md]: add json file name and explanation

File: misc/packaging/README.md (hunk 2 above)
Comment (zhanqigithub) [reply misc/packaging/README.md]: add 18.04

File: misc/packaging/README.md (hunk 4 above)
Comment (zhanqigithub) [reply misc/packaging/README.md]: if do not have superuser right will interrupt the whole install process, e.g input password.... so need this right

File: misc/packaging/README.md (hunk 6 above)
Comment (zhanqigithub) [reply misc/packaging/README.md]: add more detailed description

File: misc/packaging/README.md (hunk 7 above)
Comment (zhanqigithub) [reply misc/packaging/README.md]: user should know whether to compile

File: misc/packaging/README.md (hunk 8 above)
Comment (zhanqigithub) [reply misc/packaging/README.md]: user should know whether to sync code or not

File: misc/packaging/README.md (hunk 10 above)
Comment (zhanqigithub) [reply misc/packaging/README.md]: same as above

File: misc/packaging/README.md (hunk 11 above)
Comment (zhanqigithub) [reply misc/packaging/README.md]: must

File: misc/packaging/README.md (hunk 12 above)
Comment (zhanqigithub) [reply misc/packaging/README.md]: yes,could be all true

File: misc/packaging/README.md (hunk 13 above)
Comment (zhanqigithub) [reply misc/packaging/README.md]: yes,could be true

File: misc/packaging/README.md (hunk 14 above)
Comment (zhanqigithub) [reply misc/packaging/README.md]: add explanation

File: misc/packaging/README.md (hunk 15 above)
Comment (zhanqigithub) [reply misc/packaging/README.md]: user should know whether to build or not

File: misc/packaging/README.md (hunk 16 above)
Comment (zhanqigithub) [reply misc/packaging/README.md]: user should know whether to reboot or not

File: misc/packaging/acrn-hypervisor.postinst (hunk 17 above)
Comment (zhanqigithub) [reply misc/packaging/acrn-hypervisor.postinst]: @NanlinXie please help on this

File: misc/packaging/acrn-hypervisor.postinst (hunk 18 above)
Comment (zhanqigithub) [reply misc/packaging/acrn-hypervisor.postinst]: change typo

File: misc/packaging/README.md [hunk 21]
```diff
@@ -0,0 +1,117 @@
...
+after finished , will get below two item
+
+acrn_deb_package.deb
+
+acrn_kernel_deb_package.deb
+
+
+install command
+
+dpkg -i acrn_deb_package.deb
```
[the remark this answers is not in the document]
Comment (zhanqigithub) [reply misc/packaging/README.md]: add sudo

File: misc/packaging/README.md [hunk 22]
```diff
@@ -0,0 +1,117 @@
...
+
+dpkg -i acrn_kernel_deb_package.deb
+
+
+uninstall
+
+dpkg -r acrn-package
```
[the remark this answers is not in the document]
Comment (zhanqigithub) [reply misc/packaging/README.md]: this error seems uninstall with xxx.deb, this is not correct , should uninstall with package name not deb name

File: misc/packaging/acrn-hypervisor.postinst (hunk 20 above)
Comment (zhanqigithub) [reply misc/packaging/acrn-hypervisor.postinst:15]: next version will do error checking

File: misc/packaging/acrn-hypervisor.postinst (hunk 19 above)
Comment (zhanqigithub) [reply misc/packaging/acrn-hypervisor.postinst:30]: this version will include two boards

File: misc/packaging/README.md (hunk 9 above)
Comment (zhanqigithub) [reply misc/packaging/README.md]: this is the whole branch

**Comment by wenlingz** [comment]
@zhanqigithub make doc fail, please check:
make -C /root/workspace/acrn-hypervisor-premerge-build-image/acrn-hypervisor/doc html BUILDDIR=/root/workspace/acrn-hypervisor-premerge-build-image/acrn-hypervisor/build/doc
make[1]: Entering directory '/root/workspace/acrn-hypervisor-premerge-build-image/acrn-hypervisor/doc'

New errors/warnings found, please fix them:
==============================================

/root/workspace/acrn-hypervisor-premerge-build-image/acrn-hypervisor/build/doc/rst/misc/packaging/README.rst: WARNING: document isn't included in any toctree

make[1]: Leaving directory '/root/workspace/acrn-hypervisor-premerge-build-image/acrn-hypervisor/doc'
make[1]: *** [Makefile:61: html] Error 1
make: *** [Makefile:207: doc] Error 2

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

**Comment by wenlingz** [comment]
OK to Verify

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

**Comment by wenlingz** [comment]
@zhanqigithub  make doc fail, please refer:
hypervisor-premerge-build-image3/acrn-hypervisor/build/doc
make[1]: Entering directory '/root/workspace/acrn-hypervisor-premerge-build-image3/acrn-hypervisor/doc'

New errors/warnings found, please fix them:
==============================================

/root/workspace/acrn-hypervisor-premerge-build-image3/acrn-hypervisor/build/doc/rst/misc/packaging/README.rst:22: WARNING: Title underline too short.

2. config release.json file
==========================
/root/workspace/acrn-hypervisor-premerge-build-image3/acrn-hypervisor/build/doc/rst/misc/packaging/README.rst:22: WARNING: Title underline too short.

2. config release.json file
==========================

make[1]: *** [Makefile:61: html] Error 1
make[1]: Leaving directory '/root/workspace/acrn-hypervisor-premerge-build-image3/acrn-hypervisor/doc'
make: *** [Makefile:207: doc] Error 2

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

**Pull request merged**